lib/matplotlib/axis.py

```diff
@@ -793,9 +806,14 @@ def reset_ticks(self):
         self._lastNumMajorTicks = 1
         self._lastNumMinorTicks = 1
 
+        try:
+            self.set_clip_path(self.axes.patch)
```

Comment:
It's been such a pain having to remember to do this for polar plots, so I'm happy to see it centralized.

Reply:
OK, sorry to dredge this up:

This change is a bit strange because from a literal point of view most of the axis is drawn outside this clip_path.  I was trying to short-circuit getting tightbboxes if the clip path was the axes, but with this change axis-es are marked as clipped to the axes.  Which they aren't - only their gridlines are.  

Not sure what the solution is, but wanted to point out that this is pretty confusing.

Reply:
This isn't exactly new though; it simply makes `reset_ticks` consistent with what happens in `__init__` (admittedly, of the Axes, not the Tick).